extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlDrain.java

```diff
@@ -102,23 +104,25 @@ private void validatePods(List<V1Pod> pods) throws KubectlException {
 
   private void deletePod(CoreV1Api api, String name, String namespace)
       throws ApiException, IOException, KubectlException {
-    api.deleteNamespacedPod(name, namespace, null, null, null, null, null, null);
+    api.deleteNamespacedPod(name, namespace, null, null, this.timeoutSeconds, null, null, null);
     waitForPodDelete(api, name, namespace);
   }
 
   private void waitForPodDelete(CoreV1Api api, String name, String namespace)
-      throws ApiException, IOException, KubectlException {
+          throws KubectlException {
     long start = System.currentTimeMillis();
-    while (System.currentTimeMillis() - start < timeoutSeconds * 1000) {
+    do {
```

Comment:
replace `whilte(...){}` with `do {} whille(...)` so that kubectl drain works upon zero or minus grace-period